mesonbuild/interpreter.py

```diff
@@ -2098,10 +2098,20 @@ def func_dependency(self, node, args, kwargs):
                 if dirname in self.subprojects:
                     subproject = self.subprojects[dirname]
                     try:
-                        # Never add fallback deps to self.coredata.deps
-                        return subproject.get_variable_method([varname], {})
+                        dep = subproject.get_variable_method([varname], {})
                     except KeyError:
                         pass
+                    else:
+                        version = kwargs.get('version')
+                        dep = self.check_version(name, dep, version, dirname, varname)
+                        if dep:
+                            found_msg = ['Configured subproject dependency',
+                                mlog.bold(name), 'found:', mlog.green('YES')]
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->